### DISCUSSION TITLE
How Can I Pass Parameters When Registering a Node？

### DISCUSSION THREAD

**Discussion by sweaterlog** [BehaviorTree/BehaviorTree.CPP#700] [Q&A] [answered]
I want to pass parameters while registering a node but i don't know how to do it.
For example, in class RepeatNode i saw RepeatNode(const std::string& name, int NTries);
![image](https://github.com/BehaviorTree/BehaviorTree.CPP/assets/70195008/72f0bf49-2bf3-447f-b00e-9541594f749d)
But i don't know how to use this to construct a node.
Hopefully i can get some help.
Thanks.

**Comment by HenrikNilssonCode** [comment, marked as answer]
You can include your additional parameters as additional arguments to `factory.registerNodeType`, for example:

```
factory.registerNodeType<RepeatNode>("Repeat", 6);
```

You can also do it by way of an initializer function. Note that when passing references you have to explicitly call `std::ref`.

This is all explained in [Tutorial T08](https://www.behaviortree.dev/docs/tutorial-basics/tutorial_08_additional_args) ([Tutorial file](https://github.com/BehaviorTree/BehaviorTree.CPP/blob/master/examples/t08_additional_node_args.cpp)).
